h/api/test/auth_test.py

```diff
@@ -28,3 +31,209 @@ def test_translate_annotation_principals(p_in, p_out):
     result = auth.translate_annotation_principals(p_in)
 
     assert set(result) == set(p_out)
+
+
+generate_bearer_token_fixtures = pytest.mark.usefixtures('encode')
+
+
+@generate_bearer_token_fixtures
+def test_generate_bearer_token_encodes_userid_as_sub(encode):
+    """It should encode request.authenticated_userid as 'sub'."""
+    auth.generate_bearer_token(mock_request(), 3600)
+
+    assert encode.call_args[0][0]['sub'] == 'acct:testuser@hypothes.is'
+
+
+@generate_bearer_token_fixtures
+def test_generate_bearer_token_when_authenticated_userid_is_None(encode):
+    """It should work when request.authenticated_userid is None."""
+    request = mock_request()
+    request.authenticated_userid = None
+
+    auth.generate_bearer_token(request, 3600)
+
+    assert encode.call_args[0][0]['sub'] is None
+
+
+@generate_bearer_token_fixtures
+def test_generate_bearer_token_passes_expiry_time_to_encode(encode):
+    """It should encode the given expiry time as 'exp'."""
+    before = datetime.datetime.utcnow()
+
+    auth.generate_bearer_token(mock_request(), 3600)
+
+    after = datetime.datetime.utcnow() + datetime.timedelta(seconds=3600)
+    assert before < encode.call_args[0][0]['exp'] < after
+
+
+@generate_bearer_token_fixtures
+def test_generate_bearer_token_passes_audience_to_encode(encode):
+    """It should encode request.host_url as 'aud'."""
+    request = mock_request()
+
+    auth.generate_bearer_token(request, 3600)
+
+    assert encode.call_args[0][0]['aud'] == request.host_url
+
+
+@generate_bearer_token_fixtures
+def test_generate_bearer_token_passes_algorithm_to_encode(encode):
+    """It should encode request.host_url as 'aud'."""
```

Comment:
This docstring doesn't match the test. 

Overall I'm not sure we need to test the passing of the various claims quite so laboriously. One test should suffice to verify the claim dictionary.